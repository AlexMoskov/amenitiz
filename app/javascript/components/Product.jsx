import React from "react";
import { Link } from "react-router-dom";
import apiService from '../services/apiService';

class Product extends React.Component {
    constructor(props) {
        super(props);
        this.state = { product: { } };
    }

    componentDidMount() {
        const {
            match: {
                params: { id }
            }
        } = this.props;

        apiService.getProduct(id)
            .then(response => this.setState({ product: response }))
            .catch(() => this.props.history.push("/products"));
    }

    render() {
        const { product } = this.state;

        return (
            <React.Fragment>
                <div className="py-5">
                    <main className="container">
                        <div className="position-relative d-flex align-items-center justify-content-center">
                            <table cellPadding="7" border="2" width="100%">
                                <thead>
                                     <tr>
                                         <th>Product Code</th>
                                         <th>Product name</th>
                                         <th>Price</th>
                                     </tr>
                                </thead>
                                <tbody>
                                    <tr>
                                        <td>{product.product_code}</td>
                                        <td>{product.name}</td>
                                        <td>{product.price}&nbsp;€</td>
                                    </tr>
                                </tbody>
                            </table>
                        </div>
                        <br/>
                        <div className='alert alert-warning'>
                            <h3>Promotion from our company!</h3>
                            <p>Buy-one-get-one-free offers and green tea.</p>
                            {product.product_code == 'SR1' && <p>
                                If you buy 3 or more strawberries, the price will drop to 4.50€.
                            </p>}
                            {product.product_code == 'CF1' && <p>
                                If you buy 3 or more coffees, the price of all coffees will drop to 2/3 of the original price.
                            </p>}
                        </div>
                        <div className="container py-5">
                            <Link to="/products" className="btn btn-lg btn-secondary">
                                Back to products
                            </Link>
                        </div>
                    </main>
                </div>
            </React.Fragment>
        );
    }
}

export default Product;