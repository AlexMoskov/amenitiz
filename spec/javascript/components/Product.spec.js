import React from 'react'
import axios from 'axios'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Product from 'components/Product'

configure({ adapter: new Adapter() });
jest.mock('axios')

describe('Product', () => {
    const props = {
        match: {
            params: {
                id: 1001
            }
        }
    }
    const component = shallow(<Product {...props} />)

    it('renders Product component', () => {
        expect(toJson(component)).toMatchSnapshot()
    })
})
