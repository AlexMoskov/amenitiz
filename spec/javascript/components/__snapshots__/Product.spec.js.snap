// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product renders Product component 1`] = `
<Fragment>
  <div
    className="py-5"
  >
    <main
      className="container"
    >
      <div
        className="position-relative d-flex align-items-center justify-content-center"
      >
        <table
          border="2"
          cellPadding="7"
          width="100%"
        >
          <thead>
            <tr>
              <th>
                Product Code
              </th>
              <th>
                Product name
              </th>
              <th>
                Price
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td />
              <td />
              <td>
                <CurrencyFormat
                  allowNegative={true}
                  decimalSeparator="."
                  displayType="text"
                  fixedDecimalScale={false}
                  isAllowed={[Function]}
                  isNumericString={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  onValueChange={[Function]}
                  prefix=""
                  suffix=" â‚¬"
                  thousandSeparator={true}
                  thousandSpacing="3"
                  type="text"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <br />
      <div
        className="alert alert-warning"
      >
        <h3>
          Promotion from our company!
        </h3>
        <p>
          Buy-one-get-one-free offers and green tea.
        </p>
      </div>
      <div
        className="container py-5"
      >
        <Link
          className="btn btn-lg btn-secondary"
          to="/products"
        >
          Back to products
        </Link>
      </div>
    </main>
  </div>
</Fragment>
`;
